README_cmake.md

```diff
@@ -1,106 +1,377 @@
+repository. On WSL 2, all methods are available.
+
+## Installing dependencies
+
+We generally recommend using a package manager to fetch Halide's dependencies.
+Except where noted, we recommend using vcpkg on Windows, Homebrew on macOS, and
+apt on Ubuntu 20.04 LTS.
+
+Only LLVM and Clang are _absolutely_ required to build Halide. Halide always
+supports three LLVM versions: the current major version, the previous major
+version, and trunk. For most users, we recommend using a binary release of LLVM
+rather than building it yourself.
+
+However, to run all of the tests and apps, an extended set is needed. This
+includes lld, Python 3, libpng, libjpeg, Doxygen, OpenBLAS, ATLAS, and Eigen3.
+While not required to build any part of Halide, we find that Ninja is the best
+backend build tool across all platforms.
+
+Note that CMake has many special variables for overriding the locations of
+packages and executables. Consult the CMake documentation for these. Normally,
+you should prefer to make sure your environment is sound. For instance, if you
+want CMake to use a particular version of Python, create a virtual environment
+and activate it _before_ configuring Halide.
+
+### Windows
+
+We assume you have vcpkg installed at `D:\vcpkg`. Follow the instructions in the
+vcpkg README to install. First install LLVM via vcpkg.
+
+```
+D:\vcpkg> .\vcpkg install llvm[target-all,clang-tools-extra]:x64-windows
+D:\vcpkg> .\vcpkg install llvm[target-all,clang-tools-extra]:x86-windows
 ```
 
-You will need to add the base of the Halide installation directory to
-`CMAKE_MODULE_PATH` at the CMake command line.
+This will also install Clang and LLD. Note that this will take a long time and
+take up a lot of disk space. After it is built, it is safe to delete
+intermediate build files and caches in `%VCPKG_ROOT%\buildtrees` and
+`%APPDATA%\local\vcpkg`.
+
+Then install the other libraries:
 
 ```
-$ mkdir build
-$ cd build
-$ cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_MODULE_PATH="/path/to/Halide" ..
+D:\vcpkg> .\vcpkg install libpng:x64-windows libjpeg-turbo:x64-windows openblas:x64-windows eigen3:x64-windows
+D:\vcpkg> .\vcpkg install libpng:x86-windows libjpeg-turbo:x86-windows openblas:x86-windows eigen3:x86-windows
 ```
 
-_**Note:**_ If you are using [`vcpkg`](https://github.com/Microsoft/vcpkg), this
-is configured for you by the toolchain and you do not need to do anything
-special to use Halide.
+To build the documentation, you will need to install Doxygen. This can be done
+either through Chocolatey or from their website.
 
-### Getting CMake
+```
+> choco install doxygen
+```
 
-Getting a recent version of CMake couldn't be easier, and there are multiple
-good options on any system to do so.
+To build the Python bindings, you will need to install Python 3. This should be
+done by running the official installer from the Python website. Be sure to
+install system-wide (in `C:\Program Files`) and download the debugging symbols
```

Comment:
There's no particular requirement it goes in Program files, right? You just need to check the box that says to add it to the path.

Reply:
I vaguely recall CMake having trouble finding it installed to a non-standard location

Reply:
> I vaguely recall CMake having trouble finding it installed to a non-standard location

leave out everything after "CMake having trouble finding", it's cleaner